src/prefect/automations.py

```diff
@@ -0,0 +1,158 @@
+    ResourceTrigger,
+    SequenceTrigger,
+    Trigger,
+)
+from prefect.exceptions import PrefectHTTPStatusError
+from prefect.utilities.asyncutils import sync_compatible
+
+__all__ = [
+    "AutomationCore",
+    "EventTrigger",
+    "ResourceTrigger",
+    "Posture",
+    "Trigger",
+    "ResourceSpecification",
+    "MetricTriggerOperator",
+    "MetricTrigger",
+    "PrefectMetric",
+    "CompositeTrigger",
+    "SequenceTrigger",
+    "CompoundTrigger",
+    "MetricTriggerQuery",
+]
+
+
+class Automation(AutomationCore):
+    id: Optional[UUID] = Field(default=None, description="The ID of this automation")
+
+    @classmethod
+    @sync_compatible
+    async def create(cls: Self, automation: Self) -> Self:
+        """
+        Create a new automation.
+
+        auto_to_create = Automation(
+            name="woodchonk",
+            trigger=EventTrigger(
+                expect={"animal.walked"},
+                match={
+                    "genus": "Marmota",
+                    "species": "monax",
+                },
+                posture="Reactive",
+                threshold=3,
+                within=timedelta(seconds=10),
+            ),
+            actions=[CancelFlowRun()]
+        )
+        created_automation = Automation.create(auto_to_create)
+        """
+        client, _ = get_or_create_client()
+        automation = AutomationCore(**automation.dict(exclude={"id"}))
+        automation_id = await client.create_automation(automation=automation)
+        automation = await client.read_automation(automation_id=automation_id)
+        automation = Automation(**automation.dict())
+        return automation if automation else None
+
+    @sync_compatible
+    async def update(self: Self):
+        """
+        Updates an existing automation.
+        auto = Automation.read(id = 123)
+        auto.name = "new name"
+        auto.update()
+        """
+
+        client, _ = get_or_create_client()
+        automation = AutomationCore(**self.dict(exclude={"id", "owner_resource"}))
+        await client.update_automation(automation_id=self.id, automation=automation)
+
+    @classmethod
+    @sync_compatible
+    async def read(
+        cls: Self, id: Optional[UUID] = None, name: Optional[str] = None
+    ) -> Self:
+        """
+        Read an automation by ID or name.
+        automation = Automation.read(name="woodchonk")
+
+        or
+
+        automation = Automation.read(id=UUID("b3514963-02b1-47a5-93d1-6eeb131041cb"))
+        """
+        if id and name:
+            raise ValueError("Only one of id or name can be provided")
+        if not id and not name:
+            raise ValueError("One of id or name must be provided")
+        client, _ = get_or_create_client()
+        if id:
+            automation = await client.read_automation(automation_id=id)
+            return Automation(**automation.dict()) if automation else None
+        else:
+            automation = await client.read_automations_by_name(name=name)
```

Comment:
Perhaps we should also raise an error here if `automation` is `None`. No request error is raised in the `read_automations_by_name` method as it only filters down to an empty list if there are no automations with that name so it would need to be raised here if we do.

<details>
<pre><code>
In [2]: from prefect.automations import Automation

In [3]: Automation.read(name='nonexistent'). # no output
In [4]: Automation.read(name='exists')
Out[4]: Automation(name='exists', description='', enabled=True, trigger=MetricTrigger(type='metric', match=ResourceSpecification(__root__={'prefect.resource.id': 'prefect.flow-run.*'}), match_related=ResourceSpecification(__root__={}), posture=Posture.Metric, metric=MetricTriggerQuery(name=<PrefectMetric.successes: 'successes'>, threshold=0.9, operator=<MetricTriggerOperator.LT: '<'>, range=datetime.timedelta(seconds=60), firing_for=datetime.timedelta(seconds=60))), actions=[SuspendFlowRun(type='suspend-flow-run')], actions_on_trigger=[], actions_on_resolve=[], owner_resource=None, id=UUID('6d222a09-3c68-42d4-b019-bd331a3abb88'))
</code></pre>
</details>